portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -103,8 +100,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-  * Converts {@code toConvert} into a JSON string using GSON.
-  */
+   * Creates a Comment instance using properties from {@code comment}.
+   */
+  private Comment createComment(Entity comment) {
+    long id = comment.getKey().getId();
+    String text = (String) comment.getProperty(COMMENT_TEXT);
+    long timestamp = (long) comment.getProperty(COMMENT_TIMESTAMP);
+    String name = (String) comment.getProperty(COMMENT_NAME);
+    String nickname = (String) comment.getProperty(COMMENT_NICKNAME);
+    String imageUrl = (String) comment.getProperty(COMMENT_IMAGE_URL);
+
+    return Comment.builder().setId(id).setText(text).setTimestamp(timestamp)
+              .setName(name).setNickname(nickname).setImageUrl(imageUrl).build();
+  }
+
+  /**
+   * Returns a query sorted based on input from {@code request}.
+   */
+  private Query sortedQuery(HttpServletRequest request) {
+    String sort = request.getParameter(SORT);
+    Query query = new Query(COMMENT_ENTITY);
+
+    if (sort.equals(OLDEST_FIRST)) {
+      query.addSort(COMMENT_TIMESTAMP, SortDirection.ASCENDING);
+    } else if (sort.equals(NEWEST_FIRST)) {
+      query.addSort(COMMENT_TIMESTAMP, SortDirection.DESCENDING);
+    } else if (sort.equals(LONGEST_FIRST)) {
+      query.addSort(COMMENT_LENGTH, SortDirection.DESCENDING);
+    }
+    return query;
+  }
```

Comment:
Can I suggest a switch statement for this? I wonder if these values are enums and we need to have a case for each one, a switch statement will enforce that. If they're not either way a switch statement would be much clearer rather than branching.